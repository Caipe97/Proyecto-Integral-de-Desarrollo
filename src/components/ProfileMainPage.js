import React, { Component } from 'react';
import { Link } from "react-router-dom";
import '../pages/Profile/Profile.css';
import Header from '../components/Header';
import Footer from '../components/Footer';
import CarouselContainer from '../components/CarouselContainer';

class ProfileMainPage extends Component{
  
  componentDidMount(){
    this.props.onGetMealsFromUser(this.props.userId)
  }
  render() {
    if(this.props.meals[0]){
      return (
        <div className="contenedorPro">
          <Header userId={this.props.userId} onLogout={this.props.onLogout} history={this.props.history}/>

          <div className="contenidoPro">
            <div className="search" style={{ display: 'flex', justifyContent: 'space-around' }}>
              <div style={{ width: '80%' }}>Search</div>
              <div style={{ width: '10%', }}>
                <button  type="button" className='btn btn--primary btn--s' style={{ fontSize: '9px', backgroundColor:'white'}}  onClick={async () => { await this.props.onGetAllFoods(); this.props.history.push("/meals"); }}>Agregar Alimento</button>
              </div>
            </div>
            <div className="comida"  style={{borderRadius:'18px'}}
            // style={{ display: 'flex', justifyContent: 'space-around' }}
            >
              {/* {this.props.meals.map((meal) =>
                  <div key={meal.mealId}>
                      <p>{meal.name}</p>   
                      <p>{meal.dateEaten}</p>
                      {meal.FoodList.map((FoodListItem) => {
                        return(
                          <div key={FoodListItem.food.foodId}>
                            <p>{FoodListItem.quantity} {FoodListItem.food.name}</p>
                          </div>
                        )
                      })}
                  </div>
              )} */}
              {/* ACA COLOCAR CARUSELL */}
              <CarouselContainer meals={this.props.meals} history={this.props.history}/>
            </div>
          </div>
          <div className="sidebarPro">
            {/* <div style={{ display: 'flex', justifyContent: 'center', alignItems: 'center', height: '100%', flexDirection: 'column' }}> */}
            <div className="col3" style={{ textAlign: 'center' }}>
              <h1 className='f1'>Perfil</h1>
            </div>
            <div className="col1" style={{height:'100%'}}>
              <p>Name: {this.props.name}</p>
              <p>Surname: {this.props.surname}</p>
              <p>Email: {this.props.email}</p>
              <p>Gender: {this.props.gender}</p>
              
            </div>
            <div className="col2" style={{height:'100%'}}>
              <p>Birthday: {this.props.birthday}</p>
              <p>Weight: {this.props.weight}</p>
              <p>Height: {this.props.height}</p>
              <Link to="/resetPassword" style={{ color: 'black', marginBottom: '5%' }}>Reset your password</Link>
            </div>
          </div>
          <Footer />
        </div>
      );
    } else{
      return (
        <div className="contenedorPro">
          <Header userId={this.props.userId} onLogout={this.props.onLogout} history={this.props.history} style={{ background: '#0E4749', paddingTop: '10px', paddingBottom: '10px', top: 0 }}/>

          <div className="contenidoPro" style={{backgroundColor:'#B6E052'}}>
            <div className="search" style={{ display: 'flex', justifyContent: 'space-around' }}>
              <div style={{ width: '80%' }}>Search</div>
              <div style={{ width: '10%', }}>
                <button type="button" className='btn btn--primary btn--s' style={{ fontSize: '9px', backgroundColor:'white'}} onClick={async () => { await this.props.onGetAllFoods(); this.props.history.push("/meals"); }}>Agregar Alimento</button>
              </div>
              
            </div>
            <div className="comida" style={{ display: 'flex', justifyContent: 'space-around' }}>
            </div>
          </div>
          <div className="sidebarPro">
            {/* <div style={{ display: 'flex', justifyContent: 'center', alignItems: 'center', height: '100%', flexDirection: 'column' }}> */}
            <div className="col3" style={{ textAlign: 'center' }}>
              <h1 className='f1'>Perfil</h1>
            </div>
            <div className="col1" style={{height:'100%'}}>
              <p>Name: {this.props.name}</p>
              <p>Surname: {this.props.surname}</p>
              <p>Email: {this.props.email}</p>
              <p>Gender: {this.props.gender}</p>
              
            </div>
            <div className="col2" style={{height:'100%'}}>
              <p>Birthday: {this.props.birthday}</p>
              <p>Weight: {this.props.weight}</p>
              <p>Height: {this.props.height}</p>
              <Link to="/resetPassword" style={{ color: 'black', marginBottom: '5%' }}>Reset your password</Link>
            </div>
          </div>
          <Footer />
        </div>
      );
    }
    
  }
}

export default ProfileMainPage
