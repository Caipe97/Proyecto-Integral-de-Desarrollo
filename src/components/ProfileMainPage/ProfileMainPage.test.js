import React from 'react';
import { shallow } from 'enzyme';
import ProfileMainPage from './ProfileMainPage';

describe('ProfileMainPage tests', () => {
    const historyMock = { push: jest.fn() ,location: {state:{
        dateStart: '2021-11-01T20:09:17.409Z',
        dateEnd: '2021-11-01T20:09:17.409Z',
        foodCategoriesWithCalories: [],
        filteredFoodCategories: [],
        checkboxsList: [],
        chartData: {},
        mealsQuantity: 0,
        errorMessage: ''
    }}};
    const preventDefault = { preventDefault: jest.fn() }
    let wrapper, wrapper2;

    beforeEach(() => {
        let mockProps2 = {
            userId: 1,
                name: 'julian',
                surname: 'livrone',
                email: 'julianlivrone@gmail.com',
                password: 'asd',
                gender: 'masculino',
                birthday: '2010-01-21',
                weight: 90,
                height: 170,
                meals:[
                    {mealId: 21, name: 'queso', foodsAndQuantity: {quantity: 1, foods: {foodId: 1, name: 'Milanesa', recommendedServing: 85, caloriesPerServing: 198, createdAt: '2021-09-15T19:58:04.486Z'}}, dateEaten: '2021-09-09T00:00:00.000Z', userId: 1},
                    {mealId: 22, name: 'carne', foodsAndQuantity: {quantity: 1, foods: {foodId: 1, name: 'Milanesa', recommendedServing: 85, caloriesPerServing: 198, createdAt: '2021-09-15T19:58:04.486Z'}}, dateEaten: '2021-09-09T00:00:00.000Z', userId: 1},
                    {mealId: 23, name: 'jamon', foodsAndQuantity: {quantity: 1, foods: {foodId: 1, name: 'Milanesa', recommendedServing: 85, caloriesPerServing: 198, createdAt: '2021-09-15T19:58:04.486Z'}}, dateEaten: '2021-09-09T00:00:00.000Z', userId: 1}
                ],
                foodCategories: [],
                onGetMealsFromUser:jest.fn(),
                onGetMealsByPeriod: jest.fn(),
                onGetFoodCategories: jest.fn(),
                onGetLastYearsMeals: jest.fn(),
                onLogout: jest.fn(),
                preventDefault:  jest.fn(),
                onGetGoalsFromUser:jest.fn(),
                preventDefault:jest.fn()
          };
          let mockProps = {
            userId: 1,
            name: 'julian',
            surname: 'livrone',
            email: 'julianlivrone@gmail.com',
            password: 'asd',
            gender: 'masculino',
            birthday: '2010-01-21',
            weight: 90,
            height: 170,
            meals:[],
            foodCategories: [],
            onGetMealsFromUser:jest.fn(),
            onGetMealsByPeriod: jest.fn(),
            onGetFoodCategories: jest.fn(),
            onGetLastYearsMeals: jest.fn(),
            onLogout: jest.fn(),
            preventDefault:  jest.fn(),
            onGetGoalsFromUser:jest.fn(),
            preventDefault:jest.fn()
      };
        wrapper = shallow(<ProfileMainPage {...mockProps} history={historyMock}/>);
        wrapper2 = shallow(<ProfileMainPage {...mockProps2} history={historyMock}/>);
     })
    it('expect to render ProfileMainPage component to render with signing in sin comida',async () => {
        expect(wrapper.debug()).toMatchSnapshot();
        await wrapper.instance().componentDidMount();
        await wrapper.instance().componentDidUpdate();
    })

    it('expect to render ProfileMainPage component to render with signing in con comida', async() => {
        expect(wrapper2.debug()).toMatchSnapshot();
        await wrapper2.instance().componentDidMount();
        await wrapper2.instance().componentDidUpdate();
    })

    it('handleSubmit expect onclick', async () => {
       wrapper.find('[className="button"]').at(0).simulate('click',{ preventDefault: jest.fn()});
    })
    
   
})