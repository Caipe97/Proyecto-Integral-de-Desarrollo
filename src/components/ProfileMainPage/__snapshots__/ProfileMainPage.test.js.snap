// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileMainPage tests expect to render ProfileMainPage component to render with signing in con comida 1`] = `
<div
  className="contenedorPro"
>
  <Header
    birthday="2010-01-21"
    email="julianlivrone@gmail.com"
    foodCategories={Array []}
    gender="masculino"
    height={170}
    history={
      Object {
        "push": [MockFunction],
      }
    }
    meals={
      Array [
        Object {
          "dateEaten": "2021-09-09T00:00:00.000Z",
          "foodsAndQuantity": Object {
            "foods": Object {
              "caloriesPerServing": 198,
              "createdAt": "2021-09-15T19:58:04.486Z",
              "foodId": 1,
              "name": "Milanesa",
              "recommendedServing": 85,
            },
            "quantity": 1,
          },
          "mealId": 21,
          "name": "queso",
          "userId": 1,
        },
        Object {
          "dateEaten": "2021-09-09T00:00:00.000Z",
          "foodsAndQuantity": Object {
            "foods": Object {
              "caloriesPerServing": 198,
              "createdAt": "2021-09-15T19:58:04.486Z",
              "foodId": 1,
              "name": "Milanesa",
              "recommendedServing": 85,
            },
            "quantity": 1,
          },
          "mealId": 22,
          "name": "carne",
          "userId": 1,
        },
        Object {
          "dateEaten": "2021-09-09T00:00:00.000Z",
          "foodsAndQuantity": Object {
            "foods": Object {
              "caloriesPerServing": 198,
              "createdAt": "2021-09-15T19:58:04.486Z",
              "foodId": 1,
              "name": "Milanesa",
              "recommendedServing": 85,
            },
            "quantity": 1,
          },
          "mealId": 23,
          "name": "jamon",
          "userId": 1,
        },
      ]
    }
    name="julian"
    onGetFoodCategories={
      [MockFunction] {
        "calls": Array [
          Array [
            1,
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
    onGetLastYearsMeals={
      [MockFunction] {
        "calls": Array [
          Array [
            1,
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
    onGetMealsByPeriod={
      [MockFunction] {
        "calls": Array [
          Array [
            1,
            "Nov 07 2020",
            "Oct 07 2021",
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
    onGetMealsFromUser={
      [MockFunction] {
        "calls": Array [
          Array [
            1,
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
    onLogout={[MockFunction]}
    password="asd"
    preventDefault={[MockFunction]}
    surname="livrone"
    userId={1}
    weight={90}
  />
  <div
    className="contenidoPro"
  >
    <div
      className="comida"
    >
      <MealsSearchBar
        birthday="2010-01-21"
        email="julianlivrone@gmail.com"
        foodCategories={Array []}
        gender="masculino"
        height={170}
        history={
          Object {
            "push": [MockFunction],
          }
        }
        meals={
          Array [
            Object {
              "dateEaten": "2021-09-09T00:00:00.000Z",
              "foodsAndQuantity": Object {
                "foods": Object {
                  "caloriesPerServing": 198,
                  "createdAt": "2021-09-15T19:58:04.486Z",
                  "foodId": 1,
                  "name": "Milanesa",
                  "recommendedServing": 85,
                },
                "quantity": 1,
              },
              "mealId": 21,
              "name": "queso",
              "userId": 1,
            },
            Object {
              "dateEaten": "2021-09-09T00:00:00.000Z",
              "foodsAndQuantity": Object {
                "foods": Object {
                  "caloriesPerServing": 198,
                  "createdAt": "2021-09-15T19:58:04.486Z",
                  "foodId": 1,
                  "name": "Milanesa",
                  "recommendedServing": 85,
                },
                "quantity": 1,
              },
              "mealId": 22,
              "name": "carne",
              "userId": 1,
            },
            Object {
              "dateEaten": "2021-09-09T00:00:00.000Z",
              "foodsAndQuantity": Object {
                "foods": Object {
                  "caloriesPerServing": 198,
                  "createdAt": "2021-09-15T19:58:04.486Z",
                  "foodId": 1,
                  "name": "Milanesa",
                  "recommendedServing": 85,
                },
                "quantity": 1,
              },
              "mealId": 23,
              "name": "jamon",
              "userId": 1,
            },
          ]
        }
        name="julian"
        onGetFoodCategories={
          [MockFunction] {
            "calls": Array [
              Array [
                1,
              ],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": undefined,
              },
            ],
          }
        }
        onGetLastYearsMeals={
          [MockFunction] {
            "calls": Array [
              Array [
                1,
              ],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": undefined,
              },
            ],
          }
        }
        onGetMealsByPeriod={
          [MockFunction] {
            "calls": Array [
              Array [
                1,
                "Nov 07 2020",
                "Oct 07 2021",
              ],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": undefined,
              },
            ],
          }
        }
        onGetMealsFromUser={
          [MockFunction] {
            "calls": Array [
              Array [
                1,
              ],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": undefined,
              },
            ],
          }
        }
        onLogout={[MockFunction]}
        password="asd"
        preventDefault={[MockFunction]}
        surname="livrone"
        userId={1}
        weight={90}
      />
    </div>
  </div>
  <div
    className="dashboardPro"
  >
    <div
      className="col1"
      style={
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "column",
          "justifyContent": "center",
          "textAlign": "center",
        }
      }
    >
      <p>
        Porcentaje calórico por Categoría
      </p>
      <div
        style={
          Object {
            "alignContent": "center",
            "alignItems": "center",
            "alignSelf": "center",
            "height": "100%",
            "justifyContent": "center",
            "width": "100%",
          }
        }
      >
        <ForwardRef
          data={
            Object {
              "datasets": Array [
                Object {
                  "backgroundColor": Array [
                    "rgb(255, 99, 132)",
                    "rgb(54, 162, 235)",
                    "rgb(255, 206, 86)",
                    "rgb(75, 192, 192)",
                    "rgb(153, 102, 255)",
                    "rgb(255, 159, 64)",
                  ],
                  "data": Array [],
                  "label": "Calorias por mes",
                },
              ],
              "labels": Array [],
            }
          }
          options={
            Object {
              "labels": Object {
                "display": false,
              },
              "maintainAspectRatio": false,
              "plugins": Object {
                "legend": Object {
                  "display": true,
                },
              },
              "responsive": true,
            }
          }
        />
      </div>
    </div>
    <div
      className="col3"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <form>
        <r
          allowSameDay={false}
          customTimeInput={null}
          dateFormat="dd-MM-yyyy"
          dateFormatCalendar="LLLL yyyy"
          disabled={false}
          disabledKeyboardNavigation={false}
          dropdownMode="scroll"
          enableTabLoop={true}
          excludeScrollbar={true}
          focusSelectedMonth={false}
          monthsShown={1}
          name="date"
          nextMonthButtonLabel="Next Month"
          nextYearButtonLabel="Next Year"
          onBlur={[Function]}
          onCalendarClose={[Function]}
          onCalendarOpen={[Function]}
          onChange={[Function]}
          onClickOutside={[Function]}
          onFocus={[Function]}
          onInputClick={[Function]}
          onInputError={[Function]}
          onKeyDown={[Function]}
          onMonthChange={[Function]}
          onSelect={[Function]}
          onYearChange={[Function]}
          placeholderText="Fecha de inicio"
          preventOpenOnFocus={false}
          previousMonthButtonLabel="Previous Month"
          previousYearButtonLabel="Previous Year"
          readOnly={false}
          renderDayContents={[Function]}
          selected=""
          shouldCloseOnSelect={true}
          showFourColumnMonthYearPicker={false}
          showFullMonthYearPicker={false}
          showMonthYearPicker={false}
          showPopperArrow={true}
          showPreviousMonths={false}
          showQuarterYearPicker={false}
          showTimeInput={false}
          showTimeSelect={true}
          showTwoColumnMonthYearPicker={false}
          showYearPicker={false}
          strictParsing={false}
          style={
            Object {
              "float": "left",
            }
          }
          timeCaption="Time"
          timeInputLabel="Time"
          timeIntervals={30}
          withPortal={false}
          yearItemNumber={12}
        />
        <r
          allowSameDay={false}
          customTimeInput={null}
          dateFormat="dd-MM-yyyy"
          dateFormatCalendar="LLLL yyyy"
          disabled={false}
          disabledKeyboardNavigation={false}
          dropdownMode="scroll"
          enableTabLoop={true}
          excludeScrollbar={true}
          focusSelectedMonth={false}
          monthsShown={1}
          name="date"
          nextMonthButtonLabel="Next Month"
          nextYearButtonLabel="Next Year"
          onBlur={[Function]}
          onCalendarClose={[Function]}
          onCalendarOpen={[Function]}
          onChange={[Function]}
          onClickOutside={[Function]}
          onFocus={[Function]}
          onInputClick={[Function]}
          onInputError={[Function]}
          onKeyDown={[Function]}
          onMonthChange={[Function]}
          onSelect={[Function]}
          onYearChange={[Function]}
          placeholderText="Fecha de finalización"
          preventOpenOnFocus={false}
          previousMonthButtonLabel="Previous Month"
          previousYearButtonLabel="Previous Year"
          readOnly={false}
          renderDayContents={[Function]}
          selected=""
          shouldCloseOnSelect={true}
          showFourColumnMonthYearPicker={false}
          showFullMonthYearPicker={false}
          showMonthYearPicker={false}
          showPopperArrow={true}
          showPreviousMonths={false}
          showQuarterYearPicker={false}
          showTimeInput={false}
          showTimeSelect={true}
          showTwoColumnMonthYearPicker={false}
          showYearPicker={false}
          strictParsing={false}
          style={
            Object {
              "float": "left",
            }
          }
          timeCaption="Time"
          timeInputLabel="Time"
          timeIntervals={30}
          withPortal={false}
          yearItemNumber={12}
        />
        <button
          className="button"
          onClick={[Function]}
        >
          Buscar comidas
        </button>
        <p />
      </form>
      <p>
        Filtros
      </p>
      <WithStyles(ForwardRef(FormGroup)) />
    </div>
    <div
      className="col2"
      style={
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "column",
          "justifyContent": "center",
          "minHeight": 400,
          "textAlign": "center",
        }
      }
    >
      <p>
        Consumo calórico mensual
      </p>
      <div
        style={
          Object {
            "alignContent": "center",
            "alignItems": "center",
            "alignSelf": "center",
            "height": "100%",
            "justifyContent": "center",
            "minHeight": 300,
            "width": "90%",
          }
        }
      >
        <ForwardRef
          options={
            Object {
              "maintainAspectRatio": false,
              "plugins": Object {
                "legend": Object {
                  "display": false,
                },
              },
              "responsive": true,
            }
          }
        />
      </div>
    </div>
    <div
      className="col4"
      style={Object {}}
    >
      <div
        className="comidaa"
        style={
          Object {
            "borderRadius": "18px",
          }
        }
      >
        <p>
          Comidas más abundantes
        </p>
        <CarouselContainer
          history={
            Object {
              "push": [MockFunction],
            }
          }
          meals={
            Array [
              Object {
                "dateEaten": "2021-09-09T00:00:00.000Z",
                "foodsAndQuantity": Object {
                  "foods": Object {
                    "caloriesPerServing": 198,
                    "createdAt": "2021-09-15T19:58:04.486Z",
                    "foodId": 1,
                    "name": "Milanesa",
                    "recommendedServing": 85,
                  },
                  "quantity": 1,
                },
                "mealId": 21,
                "name": "queso",
                "userId": 1,
              },
              Object {
                "dateEaten": "2021-09-09T00:00:00.000Z",
                "foodsAndQuantity": Object {
                  "foods": Object {
                    "caloriesPerServing": 198,
                    "createdAt": "2021-09-15T19:58:04.486Z",
                    "foodId": 1,
                    "name": "Milanesa",
                    "recommendedServing": 85,
                  },
                  "quantity": 1,
                },
                "mealId": 22,
                "name": "carne",
                "userId": 1,
              },
              Object {
                "dateEaten": "2021-09-09T00:00:00.000Z",
                "foodsAndQuantity": Object {
                  "foods": Object {
                    "caloriesPerServing": 198,
                    "createdAt": "2021-09-15T19:58:04.486Z",
                    "foodId": 1,
                    "name": "Milanesa",
                    "recommendedServing": 85,
                  },
                  "quantity": 1,
                },
                "mealId": 23,
                "name": "jamon",
                "userId": 1,
              },
            ]
          }
        />
      </div>
    </div>
  </div>
  <div
    className="sidebarPro"
  >
    <div
      className="col3"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <h1
        className="f1"
        style={
          Object {
            "color": "rgb(0, 38, 38)",
            "fontFamily": "Arial",
          }
        }
      >
        Perfil 
        <img
          alt="profile"
          src="avatar.png"
          style={
            Object {
              "width": 50,
            }
          }
        />
      </h1>
    </div>
    <div
      className="col1"
      style={
        Object {
          "height": "100%",
        }
      }
    >
      <p>
        Nombre: 
        julian
      </p>
      <p>
        Apellido: 
        livrone
      </p>
      <p>
        Email: 
        julianlivrone@gmail.com
      </p>
      <p>
        Genero: 
        masculino
      </p>
    </div>
    <div
      className="col2"
      style={
        Object {
          "height": "100%",
        }
      }
    >
      <p>
        Fecha de Nacimiento: 
        2010-01-21
      </p>
      <p>
        Peso: 
        90
      </p>
      <p>
        Altura: 
        170
      </p>
      <Link
        style={
          Object {
            "color": "black",
            "marginBottom": "5%",
          }
        }
        to="/resetPassword"
      >
        Cambia tu contraseña
      </Link>
    </div>
  </div>
  <Footer />
</div>
`;

exports[`ProfileMainPage tests expect to render ProfileMainPage component to render with signing in sin comida 1`] = `
<div
  className="contenedorPro"
>
  <Header
    birthday="2010-01-21"
    email="julianlivrone@gmail.com"
    foodCategories={Array []}
    gender="masculino"
    height={170}
    history={
      Object {
        "push": [MockFunction],
      }
    }
    meals={Array []}
    name="julian"
    onGetFoodCategories={
      [MockFunction] {
        "calls": Array [
          Array [
            1,
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
    onGetLastYearsMeals={
      [MockFunction] {
        "calls": Array [
          Array [
            1,
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
    onGetMealsByPeriod={
      [MockFunction] {
        "calls": Array [
          Array [
            1,
            "Nov 07 2020",
            "Oct 07 2021",
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
    onGetMealsFromUser={
      [MockFunction] {
        "calls": Array [
          Array [
            1,
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
    onLogout={[MockFunction]}
    password="asd"
    preventDefault={[MockFunction]}
    surname="livrone"
    userId={1}
    weight={90}
  />
  <div
    className="contenidoPro"
  >
    <div
      className="comida"
    >
      <MealsSearchBar
        birthday="2010-01-21"
        email="julianlivrone@gmail.com"
        foodCategories={Array []}
        gender="masculino"
        height={170}
        history={
          Object {
            "push": [MockFunction],
          }
        }
        meals={Array []}
        name="julian"
        onGetFoodCategories={
          [MockFunction] {
            "calls": Array [
              Array [
                1,
              ],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": undefined,
              },
            ],
          }
        }
        onGetLastYearsMeals={
          [MockFunction] {
            "calls": Array [
              Array [
                1,
              ],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": undefined,
              },
            ],
          }
        }
        onGetMealsByPeriod={
          [MockFunction] {
            "calls": Array [
              Array [
                1,
                "Nov 07 2020",
                "Oct 07 2021",
              ],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": undefined,
              },
            ],
          }
        }
        onGetMealsFromUser={
          [MockFunction] {
            "calls": Array [
              Array [
                1,
              ],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": undefined,
              },
            ],
          }
        }
        onLogout={[MockFunction]}
        password="asd"
        preventDefault={[MockFunction]}
        surname="livrone"
        userId={1}
        weight={90}
      />
    </div>
  </div>
  <div
    className="dashboardPro"
  >
    <div
      className="col1"
      style={
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "column",
          "justifyContent": "center",
          "textAlign": "center",
        }
      }
    >
      <p>
        Porcentaje calórico por Categoría
      </p>
      <div
        style={
          Object {
            "alignContent": "center",
            "alignItems": "center",
            "alignSelf": "center",
            "height": "100%",
            "justifyContent": "center",
            "width": "100%",
          }
        }
      >
        <ForwardRef
          data={
            Object {
              "datasets": Array [
                Object {
                  "backgroundColor": Array [
                    "rgb(255, 99, 132)",
                    "rgb(54, 162, 235)",
                    "rgb(255, 206, 86)",
                    "rgb(75, 192, 192)",
                    "rgb(153, 102, 255)",
                    "rgb(255, 159, 64)",
                  ],
                  "data": Array [],
                  "label": "Calorias por mes",
                },
              ],
              "labels": Array [],
            }
          }
          options={
            Object {
              "labels": Object {
                "display": false,
              },
              "maintainAspectRatio": false,
              "plugins": Object {
                "legend": Object {
                  "display": true,
                },
              },
              "responsive": true,
            }
          }
        />
      </div>
    </div>
    <div
      className="col3"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <form>
        <r
          allowSameDay={false}
          customTimeInput={null}
          dateFormat="dd-MM-yyyy"
          dateFormatCalendar="LLLL yyyy"
          disabled={false}
          disabledKeyboardNavigation={false}
          dropdownMode="scroll"
          enableTabLoop={true}
          excludeScrollbar={true}
          focusSelectedMonth={false}
          monthsShown={1}
          name="date"
          nextMonthButtonLabel="Next Month"
          nextYearButtonLabel="Next Year"
          onBlur={[Function]}
          onCalendarClose={[Function]}
          onCalendarOpen={[Function]}
          onChange={[Function]}
          onClickOutside={[Function]}
          onFocus={[Function]}
          onInputClick={[Function]}
          onInputError={[Function]}
          onKeyDown={[Function]}
          onMonthChange={[Function]}
          onSelect={[Function]}
          onYearChange={[Function]}
          placeholderText="Fecha de inicio"
          preventOpenOnFocus={false}
          previousMonthButtonLabel="Previous Month"
          previousYearButtonLabel="Previous Year"
          readOnly={false}
          renderDayContents={[Function]}
          selected=""
          shouldCloseOnSelect={true}
          showFourColumnMonthYearPicker={false}
          showFullMonthYearPicker={false}
          showMonthYearPicker={false}
          showPopperArrow={true}
          showPreviousMonths={false}
          showQuarterYearPicker={false}
          showTimeInput={false}
          showTimeSelect={true}
          showTwoColumnMonthYearPicker={false}
          showYearPicker={false}
          strictParsing={false}
          style={
            Object {
              "float": "left",
            }
          }
          timeCaption="Time"
          timeInputLabel="Time"
          timeIntervals={30}
          withPortal={false}
          yearItemNumber={12}
        />
        <r
          allowSameDay={false}
          customTimeInput={null}
          dateFormat="dd-MM-yyyy"
          dateFormatCalendar="LLLL yyyy"
          disabled={false}
          disabledKeyboardNavigation={false}
          dropdownMode="scroll"
          enableTabLoop={true}
          excludeScrollbar={true}
          focusSelectedMonth={false}
          monthsShown={1}
          name="date"
          nextMonthButtonLabel="Next Month"
          nextYearButtonLabel="Next Year"
          onBlur={[Function]}
          onCalendarClose={[Function]}
          onCalendarOpen={[Function]}
          onChange={[Function]}
          onClickOutside={[Function]}
          onFocus={[Function]}
          onInputClick={[Function]}
          onInputError={[Function]}
          onKeyDown={[Function]}
          onMonthChange={[Function]}
          onSelect={[Function]}
          onYearChange={[Function]}
          placeholderText="Fecha de finalización"
          preventOpenOnFocus={false}
          previousMonthButtonLabel="Previous Month"
          previousYearButtonLabel="Previous Year"
          readOnly={false}
          renderDayContents={[Function]}
          selected=""
          shouldCloseOnSelect={true}
          showFourColumnMonthYearPicker={false}
          showFullMonthYearPicker={false}
          showMonthYearPicker={false}
          showPopperArrow={true}
          showPreviousMonths={false}
          showQuarterYearPicker={false}
          showTimeInput={false}
          showTimeSelect={true}
          showTwoColumnMonthYearPicker={false}
          showYearPicker={false}
          strictParsing={false}
          style={
            Object {
              "float": "left",
            }
          }
          timeCaption="Time"
          timeInputLabel="Time"
          timeIntervals={30}
          withPortal={false}
          yearItemNumber={12}
        />
        <button
          className="button"
          onClick={[Function]}
        >
          Buscar comidas
        </button>
        <p />
      </form>
      <p>
        Filtros
      </p>
      <WithStyles(ForwardRef(FormGroup)) />
    </div>
    <div
      className="col2"
      style={
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "column",
          "justifyContent": "center",
          "minHeight": 400,
          "textAlign": "center",
        }
      }
    >
      <p>
        Consumo calórico mensual
      </p>
      <div
        style={
          Object {
            "alignContent": "center",
            "alignItems": "center",
            "alignSelf": "center",
            "height": "100%",
            "justifyContent": "center",
            "minHeight": 300,
            "width": "90%",
          }
        }
      >
        <ForwardRef
          options={
            Object {
              "maintainAspectRatio": false,
              "plugins": Object {
                "legend": Object {
                  "display": false,
                },
              },
              "responsive": true,
            }
          }
        />
      </div>
    </div>
    <div
      className="col4"
      style={Object {}}
    >
      <div
        className="comidaa"
        style={
          Object {
            "borderRadius": "18px",
          }
        }
      >
        <p>
          Comidas más abundantes
        </p>
      </div>
    </div>
  </div>
  <div
    className="sidebarPro"
  >
    <div
      className="col3"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <h1
        className="f1"
        style={
          Object {
            "color": "rgb(0, 38, 38)",
            "fontFamily": "Arial",
          }
        }
      >
        Perfil 
        <img
          alt="profile"
          src="avatar.png"
          style={
            Object {
              "width": 50,
            }
          }
        />
      </h1>
    </div>
    <div
      className="col1"
      style={
        Object {
          "height": "100%",
        }
      }
    >
      <p>
        Nombre: 
        julian
      </p>
      <p>
        Apellido: 
        livrone
      </p>
      <p>
        Email: 
        julianlivrone@gmail.com
      </p>
      <p>
        Genero: 
        masculino
      </p>
    </div>
    <div
      className="col2"
      style={
        Object {
          "height": "100%",
        }
      }
    >
      <p>
        Fecha de Nacimiento: 
        2010-01-21
      </p>
      <p>
        Peso: 
        90
      </p>
      <p>
        Altura: 
        170
      </p>
      <Link
        style={
          Object {
            "color": "black",
            "marginBottom": "5%",
          }
        }
        to="/resetPassword"
      >
        Cambia tu contraseña
      </Link>
    </div>
  </div>
  <Footer />
</div>
`;
